packages/flutter_tools/lib/src/windows/visual_studio.dart

```diff
@@ -294,22 +299,50 @@ class VisualStudio {
         ...requirementArguments,
       ], encoding: encoding);
       if (whereResult.exitCode == 0) {
-        final List<Map<String, dynamic>> installations =
-            (json.decode(whereResult.stdout) as List<dynamic>).cast<Map<String, dynamic>>();
-        if (installations.isNotEmpty) {
+        final List<Map<String, dynamic>>? installations = _tryDecodeVswhereJson(whereResult.stdout);
+        if (installations != null && installations.isNotEmpty) {
           return VswhereDetails.fromJson(validateRequirements, installations[0]);
         }
       }
     } on ArgumentError {
       // Thrown if vswhere doesn't exist; ignore and return null below.
     } on ProcessException {
       // Ignored, return null below.
-    } on FormatException {
-      // may be thrown if invalid JSON is returned.
     }
     return null;
   }
 
+  List<Map<String, dynamic>>? _tryDecodeVswhereJson(String vswhereJson) {
+    late List<dynamic> result;
```

Comment:
nit. make these both nullable instead.

For the most part, other than in tests where a field is guaranteed to be assigned in a `setUp` before referenced, avoid using `late` in the tool. `late` overrides static type checks in favor of runtime exceptions; this code looks correct now, but across a future refactor one of these fields might get referenced in a branch that never initialized them, and if the fields are `late` the compiler/analyzer can't help us.

Reply:
Thanks for the background, I wasn't aware of those differences. Fixed!